mongo/options/aggregateoptions.go

```diff
@@ -34,10 +34,18 @@ type AggregateOptions struct {
 
 	// The maximum amount of time that the query can run on the server. The default value is nil, meaning that there
 	// is no time limit for query execution.
+	//
+	// Deprecated: This option is deprecated and will eventually be removed in version 2.0 of the driver. The more general
+	// Timeout option should be used in its place to control the amount of time that the Aggregate operation can run before
+	// returning an error. MaxTime is still usable through the deprecated setter.
 	MaxTime *time.Duration
 
 	// The maximum amount of time that the server should wait for new documents to satisfy a tailable cursor query.
 	// This option is only valid for MongoDB versions >= 3.2 and is ignored for previous server versions.
+	//
+	// Deprecated: This option is deprecated and will eventually be removed in version 2.0 of the driver. The more general
+	// Timeout option should be used in its place to control the amount of time that the Aggregate operation can run before
+	// returning an error. MaxAwaitTime is still usable through the deprecated setter.
```

Comment:
MaxAwaitTime may not be needed to be deprecated. Please see comments on https://github.com/mongodb/mongo-go-driver/pull/974#discussion_r889367147.

Reply:
Great point. Done; and the error from setting `Timeout` and `MaxAwaitTime` will be done as part of GODRIVER-2342.